In this section, the pre-processing of raw reads was explained with separate .md file for each step.

The for loops are used for one-step processing for all samples with an order.

Activation of conda in the environment you are using in command line is highly recommended before proceeding.

The order of pre-processing to the alignment is given below:

1. ([PreCut.md])(<https://github.com/dincaslan/SingleCell_Experience/blob/master/scTotalRNAseqAnalysisApproaches/PreProcessing/PreCut.md>))
2. ([inSilicorRNARemoval.md])(<https://github.com/dincaslan/SingleCell_Experience/blob/master/scTotalRNAseqAnalysisApproaches/PreProcessing/inSilicorRNARemoval.md>))
3. ([PreProcessingCut.md])(<https://github.com/dincaslan/SingleCell_Experience/blob/master/scTotalRNAseqAnalysisApproaches/PreProcessing/PreProcessingCut.md>))

Alternatively, you can combine step 1 and 3 in a single step and proceed for step 2.
